### PR TITLE
std.zip additional documentation

### PR DESCRIPTION
**Pull request by davideagen**
Document the process to create a zip file.

### CODE REVIEW

File: std/zip.d [hunk 1]
```diff
@@ -38,6 +39,26 @@ void main(string[] args)
...
     }
 }
+
+// Create and write new zip file.
+import std.file: write;
+import std.string: representation;
+
+void main(string[] args)
+{
+    char[] data = cast(char[])"Test data.\n";
```
Comment (yebblies) [inline std/zip.d]: This is unsafe and probably shouldn't be in an example.  Maybe use `"...".dup` to create a mutable string?

File: std/zip.d [hunk 2]
```diff
@@ -38,6 +39,26 @@ void main(string[] args)
         assert(am.expandedData.length == am.expandedSize);
     }
 }
+
+// Create and write new zip file.
+import std.file: write;
+import std.string: representation;
+
+void main(string[] args)
```
Comment (yebblies) [inline std/zip.d]: Better to have the signature as `void main()` if `args` isn't used.

**Comment by davideagen** [comment]
I just realized I have a messy merge history here. I've opened a new pull request with a clean one. 

Thanks Yebblies, I'll fix what you mentioned in the clean PR.

**Comment by yebblies** [comment]
This will need to be rebased to remove the merge commits.

**Comment by yebblies** [comment]
Ok oh.  You don't need to open a new PR, you can just force push to the original branch and the pull request will update.

**Comment by davideagen** [comment]
Github won't let me reopen after force-pushing so I have to go with the new PR now. But at least I now have a better understanding of how to handle this situation next time.

**Comment by yebblies** [comment]
Oh yeah, you have to remember to reopen before force-pushing.

**Pull request closed without merging**